server/notebooks/tasks.py

```diff
@@ -0,0 +1,46 @@
+import pytz
+from datetime import datetime, timedelta
+from collections import defaultdict
+from spinach import Tasks
+
+from .models import NotebookRevision
+
+TIME_INTERVAL_IN_MINS = 1
+tasks = Tasks()
+
+
+@tasks.task(name="notebooks:execute_notebook_revisions_cleanup")
+def execute_notebook_revisions_cleanup(notebook_id, now_utc=None):
+    "Prue revision hisotry."
+
+    def group_by_time_interval(notebook_revisions, time_interval_in_mins=TIME_INTERVAL_IN_MINS):
+        "Split notebook revisions into groups by time_interval_in_mins."
+        revision_groups = defaultdict(list)
+        for revision in notebook_revisions:
+            key = int(revision.created.timestamp()) // (time_interval_in_mins * 60)
+            revision_groups[key].append(revision)
+        return [revision_groups[k] for k in sorted(revision_groups.keys())]
+
+    def clear_draft_flag(notebook_revisions, now_utc, threshold_in_mins=TIME_INTERVAL_IN_MINS):
+        "Set is_draft flag to False if the revision is older than the threshold_in_mins."
+        for revision in notebook_revisions:
+            if now_utc - revision.created > timedelta(minutes=threshold_in_mins):
+                revision.is_draft = False
+                revision.save()
+
+    def delete(notebook_revisions, now_utc, threshold_in_mins=TIME_INTERVAL_IN_MINS):
+        "Delete the revision if it is older than threshold_in_mins."
+        for revision in notebook_revisions:
+            if now_utc - revision.created > timedelta(minutes=threshold_in_mins):
+                revision.delete()
```

Comment:
thanks, I understand multiple transactions are slow. I thought in current settings (30s in UI, 1 minute interval in backend), normally only one revision need to be deleted. So I didn't try to optimize here. But since in the future, we might want to have more frequent drafts, so it might worth to add this now.

I also thought about create a partial index on `is_draft`, because usually draft revisions are just a small part of all revisions, but I prefer to do that after we have some performance data.